### PR TITLE
Add api docs for `src-choice-card`

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- break out components into individual files
- adds API docs
- adds some extra stories to cover combinations of props
- ~~marks `defaultChecked` as deprecated (is this ok @SiAdcock?)~~

## Screenshots

image

### CODE REVIEW

**Comment by sndrs** [comment]
https://github.com/guardian/source/runs/3313666412#step:4:31 is being thrown because storybook needs the react import to handle a fragment, but it _shouldn't_ because TS is configured to obviate it. 

i'm not sure how to get storybook to chillout? 

maybe switch to ts-loader?

File: src/core/components/choice-card/ChoiceCardGroup.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+	argTypes: {
+		columns: {
+			options: [undefined, 2, 3, 4, 5],
+			control: { type: 'select' },
+		},
+		error: {
+			options: [undefined, 'example error message'],
+			control: { type: 'select' },
+		},
+	},
```
Comment (jamie-lynch) [inline src/core/components/choice-card/ChoiceCardGroup.stories.tsx:73]: Can we disable the `cssOverrides` control as it doesn't work in storybook?

```suggestion
	argTypes: {
		columns: {
			options: [undefined, 2, 3, 4, 5],
			control: { type: 'select' },
		},
		error: {
			options: [undefined, 'example error message'],
			control: { type: 'select' },
		},
		cssOverrides: {
			control: { disable: true },
		},
	},
```

Comment (sndrs) [reply src/core/components/choice-card/ChoiceCardGroup.stories.tsx:73]: as discussed offline, will do for all chromaticStories in separate PR

File: src/core/components/choice-card/ChoiceCard.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			control: { disable: true },
+		},
+		icon: {
+			options: ['undefined', 'JSX element'],
+			mapping: {
+				undefined: undefined,
+				'JSX element': <SvgCamera />,
+			},
+			control: { type: 'radio' },
+		},
```
[the remark this answers is not in the document]
Comment (sndrs) [reply src/core/components/choice-card/ChoiceCard.stories.tsx:36]: discussed offline, jamie will pick this up

[the remark this answers is not in the document]
Comment (jamie-lynch) [reply src/core/components/choice-card/ChoiceCard.stories.tsx:36]: Done in https://github.com/guardian/source/pull/933

File: src/core/components/choice-card/ChoiceCard.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	title: 'Source/src-choice-card/ChoiceCard',
+	component: ChoiceCard,
+	args: {
+		id: 'option-1-id',
+		value: 'option-1',
+		label: 'string',
+	},
+	argTypes: {
+		// include all args so they render in the correct order
+		id: undefined,
```
[the remark this answers is not in the document]
Comment (sndrs) [reply src/core/components/choice-card/ChoiceCard.stories.tsx:16]: also disussed offline – only needed to keep prop order in the story, so lets go with `undefined` as it's more accurate

**Review by SiAdcock — APPROVED** [review]
This looks awesome 💯

**Comment by SiAdcock** [comment]
Finally a passing build. I'll tidy up the commit history and get this merged. I've created a card to address the [Storybook Fragment issue](https://trello.com/c/Iuh3to0y/442-fix-issue-with-fragments-in-stories)

**Pull request merged**